index.d.ts

```diff
@@ -107,6 +107,7 @@ export interface AxiosRequestConfig<D = any> {
   transitional?: TransitionalOptions;
   signal?: AbortSignal;
   insecureHTTPParser?: boolean;
+  customConfig?: Record<string, any>;
```

Comment:
I don't really like the naming being forced, is it not possible to allow keys and values to be sent direclty without a set key? I feel this would make the request config interface simpler to use and mutate as the user wishes to a degree.